Problem: block param is not aligned with go-ethereum in debug_traceCall

mmsqe opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

Closes: #XXX

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

mmsqe committed:
test_tracecall_insufficient_funds

mmsqe committed:
test_tracecall_prestate_tracer

mmsqe committed:
test_debug_tracecall_call_tracer

mmsqe committed:
test_debug_tracecall_state_overrides

mmsqe committed:
Problem: block param is not aligned with go-ethereum in debug_traceCall

mmsqe committed:
test_tracecall_struct_tracer

mmsqe committed:
test_custom_js_tracers

mmsqe commented on CHANGELOG.md:
```diff
@@ -69,7 +69,8 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (rpc) [#386](https://github.com/crypto-org-chain/ethermint/pull/386) Cleanup unused cancel function in filter.
- * (rpc) [#388](https://github.com/crypto-org-chain/ethermint/pull/388) Avoid out of bound panic when error message.
+* (rpc) [#388](https://github.com/crypto-org-chain/ethermint/pull/388) Avoid out of bound panic when error message.
+* (rpc) [#](https://github.com/crypto-org-chain/ethermint/pull/) Align block param with go-ethereum in debug_traceCall.
```
```suggestion
* (rpc) [#391](https://github.com/crypto-org-chain/ethermint/pull/391) Align block param with go-ethereum in debug_traceCall.
```

mmsqe committed:
Update CHANGELOG.md

Signed-off-by: mmsqe <mavis@crypto.com>

yihuang commented on rpc/backend/tracing.go:
```diff
@@ -246,7 +246,7 @@ func (b *Backend) TraceCall(
 
 	// minus one to get the context of block beginning
-	contextHeight := blk.Block.Height - 1
+	contextHeight := blk.Block.Height
```
the minus one is to use the state of the beginning of the block, do you just want to change the block number passed to evm context?

mmsqe replied:
this is to avoid [w3_wait_for_new_blocks(w3, 1, sleep=0.1)](https://github.com/crypto-org-chain/ethermint/pull/391/files#diff-3e148d436d65c5cd83194e9ed01b8f4c97931ce2410fe2bdde6e840b6f14965cL437) before and won't use [req.BlockNumber](https://github.com/crypto-org-chain/ethermint/blob/develop/x/evm/keeper/grpc_query.go#L551) at last?

mmsqe committed:
Apply suggestions from code review

for more info, https://github.com/ethereum/go-ethereum/blob/release/1.11/eth/tracers/api.go#L919
